### PR TITLE
Add WebSockets support + ApiV2

### PR DESCRIPTION
**Pull request by aoudiamoncef**
* [x] document all added functions
* [x] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

### CODE REVIEW

File: massa-consensus-worker/src/worker/mod.rs [hunk 1]
```diff
@@ -35,6 +36,18 @@ pub struct ConsensusWorker {
...
+pub struct WsConfig {
+    /// Whether WebSockets are enabled
+    pub enabled: bool,
+    /// Broadcast sender(channel) for blocks headers
+    pub block_header_sender: Sender<BlockHeader>,
+    /// Broadcast sender(channel) for blocks
+    pub block_sender: Sender<Block>,
+    /// Broadcast sender(channel) for filled blocks
+    pub filled_block_sender: Sender<FilledBlock>,
+}
```
Comment (Eitu33) [inline massa-consensus-worker/src/worker/mod.rs:49]: Was there any discussion I'm unaware of where it was decided that we'd use tokio channels for this? Why not use crossbeam instead?

File: massa-consensus-worker/src/controller.rs [hunk 2]
```diff
@@ -219,6 +227,28 @@ impl ConsensusController for ConsensusControllerImpl {
...
+                        }
+                    })
+                    .collect();
+
+                let _ = self.ws_config.block_sender.send(value.content.clone());
+                self.ws_config.filled_block_sender.send(FilledBlock {
+                    header: value.content.header.clone(),
+                    operations,
+                })
+            });
```
Comment (Eitu33) [inline massa-consensus-worker/src/controller.rs:249]: You are using `map` to produce side effects, `map` should only be used when it is needed to modify the value and wield its result, please use `if let Some(value)` here

File: massa-consensus-worker/src/worker/mod.rs (hunk 1 above)
Comment (aoudiamoncef) [reply massa-consensus-worker/src/worker/mod.rs:49]: IMHO the new library is built on top of Tokio so it seems to be a safe choice.

this is the [example](https://github.com/paritytech/jsonrpsee/blob/master/examples/examples/ws_pubsub_broadcast.rs) which I followed to dev this feature.

https://github.com/massalabs/massa/issues/3076

File: massa-models/src/config/constants.rs [hunk 3]
```diff
@@ -64,7 +64,7 @@ lazy_static::lazy_static! {
         if cfg!(feature = "sandbox") {
             "SAND.0.0"
         } else {
-            "TEST.18.0"
+            "TEST.17.2"
```
Comment (AurelienFT) [inline massa-models/src/config/constants.rs:67]: Shouldn't be push I think.

File: massa-node/base_config/config.toml [hunk 4]
```diff
@@ -11,6 +11,8 @@
     bind_private = "127.0.0.1:33034"
     # port on which the node API listens for public requests. Can be exposed to the Internet
     bind_public = "0.0.0.0:33035"
+    # port on which the node API listens for HTTP requests and WebSOckets subscriptions. Can be exposed to the Internet
+    bind_api = "0.0.0.0:33036"
```
Comment (AurelienFT) [inline massa-node/base_config/config.toml:15]: Maybe rename it to v2 to be more clear for users.

Comment (aoudiamoncef) [reply massa-node/base_config/config.toml:15]: Just add it in the comment could be fine, I don't went to add v2 everywhere

Comment (AurelienFT) [reply massa-node/base_config/config.toml:15]: Adding it in config file that users reads doesn't seems too much to me but you can just put a comment ok

File: massa-models/src/config/constants.rs (hunk 3 above)
Comment (aoudiamoncef) [reply massa-models/src/config/constants.rs:67]: Good catch

File: massa-node/base_config/config.toml (hunk 4 above)
Comment (aoudiamoncef) [reply massa-node/base_config/config.toml:15]: Done

File: massa-models/src/config/constants.rs (hunk 3 above)
Comment (aoudiamoncef) [reply massa-models/src/config/constants.rs]: Done

**Comment by aoudiamoncef** [comment]
Please don't merge this until all the PR checkboxes are filled. We should write OpenRPC doc

**Review by AurelienFT — APPROVED** [review]

**Pull request merged**